src/core/analysis/stoppingRule/GelmanRubinStoppingRule.cpp

```diff
@@ -104,7 +104,7 @@ bool GelmanRubinStoppingRule::stop( size_t g )
         num_variables_in_rep = data[i].size();
         
         // if we have different numbers of variables in different replicates, something has gone terribly wrong
-        if (i > 1 && data[i].size() != num_variables_in_rep)
+        if (i > 0 && data[i].size() != num_variables_in_rep)
```

Comment:
Line 104 has just defined `num_variables_in_rep = data[i].size();` so I don't think this check can fail.

Perhaps it is intended to come before the previous line, or, equivalently, to compare with `data[i-1].size()`?

Reply:
As a secondary thought, defining `num_variables_in_rep` at the start of the loop would allow us to replace the calls to `data[i].size()` in each loop heading with a reference to the variable.

Reply:
Duh! What was I thinking? Thank you for spotting this!

Reply:
And yes, this could be solved in a number of ways. `i > 0 && data[i-1].size() != num_variables_in_rep` is one option. Defining `num_variables_in_rep` outside the loop is another, but then the trace-reading code would be duplicated both inside and outside the loop. Maybe the simplest solution is to redefine `num_variables_in_rep` _after_ the check, rather than before.